lib/js/components/ProgressDonutChart/ProgressDonutChart.vue

```diff
@@ -0,0 +1,287 @@
+<template>
+	<div
+		:class="{
+			progressDonutChart: true,
+		}"
+	>
+		<svg
+			class="progressDonutChart__svg"
+			width="40"
+			height="40"
+			xmlns="http://www.w3.org/2000/svg"
+		>
+			<circle
+				class="progressDonutChart__circle progressDonutChart__thumb"
+				cx="20"
+				cy="20"
+				r="18"
+			/>
+			<circle
+				v-if="state === PROGRESS_DONUT_CHART_STATES.LOADING"
+				class="progressDonutChart__circle progressDonutChart__loader"
+				cx="20"
+				cy="20"
+				r="18"
+			/>
+			<template v-else>
+				<circle
+					v-for="(range, index) in calculatedRanges"
+					:key="`circle_${index}`"
+					class="progressDonutChart__circle progressDonutChart__track"
+					cx="20"
+					cy="20"
+					r="18"
+					:style="`--length: ${range.length}; transform: rotate(calc(90deg + ${range.rotate}deg));`"
```

Comment:
should we add a unit test?

Reply:
[`72ec278` (#274)](https://github.com/bethinkpl/design-system/pull/274/commits/72ec278a74c1d268b6cbb5fa8606f72dcfa0b9c3)